astropy/utils/data.py

```diff
@@ -104,14 +104,20 @@ class CacheMissingWarning(AstropyWarning):
     """
 
 
-def _is_url(string):
+def is_url(string):
     """
-    Test whether a string is a valid URL
+    Test whether a string is a valid URL.
```

Comment:
I think we need to be clearer about what this actually tests if it's going to be a public function. Is it just URLs acceptable to `urllib.request`? IANA URLs? Will it ever raise an exception for a suitably malformed string?

Reply:
Clarified in the docstring.